Source/dead.cpp

```diff
@@ -17,25 +17,22 @@ int8_t stonendx;
 
 void InitDead()
 {
-	int8_t mtypes[MAXMONSTERS];
-
-	for (int8_t &mtype : mtypes)
-		mtype = 0;
+	int8_t mtypes[MAXMONSTERS] = {};
 
 	int8_t nd = 0;
 
 	for (int i = 0; i < nummtypes; i++) {
-		if (mtypes[Monsters[i].mtype] == 0) {
-			for (int d = 0; d < 8; d++)
-				dead[nd]._deadData[d] = Monsters[i].Anims[MA_DEATH].Data[d];
-			dead[nd]._deadFrame = Monsters[i].Anims[MA_DEATH].Frames;
-			dead[nd]._deadWidth = Monsters[i].width;
-			dead[nd]._deadtrans = 0;
-			nd++;
+		if (mtypes[Monsters[i].mtype] != 0)
+			continue;
 
-			Monsters[i].mdeadval = nd;
-			mtypes[Monsters[i].mtype] = nd;
-		}
+		dead[nd]._deadData = Monsters[i].Anims[MA_DEATH].Data;

```

Comment:
Is this really equivalent to the old 8 position loop?

Reply:
yes, since the type was changed.